src/main/java/org/codehaus/groovy/runtime/InvokerHelper.java

```diff
@@ -396,10 +402,33 @@ public static List createList(Object[] values) {
         return answer;
     }
 
+    /**
+     * Get the initial capacity of hash map, which is the closest power of 2 to the entry count.
+     * (SEE https://stackoverflow.com/questions/8352378/why-does-hashmap-require-that-the-initial-capacity-be-a-power-of-two)
+     *
+     * e.g.
+     * 1: 1
+     * 2: 2
+     * 3: 4
+     * 4: 4
+     * 5: 8
+     * 6: 8
+     * 7: 8
+     * 8: 8
+     * 9: 16
+     * ...
+     *
+     * @param initialEntryCnt the initial entry count
+     * @return the initial capacity
+     */
+    public static int initialCapacity(int initialEntryCnt) {
+        return (int) pow(2, ceil(log(initialEntryCnt) / LN2));
```

Comment:
How about  return 2<<Integer.highestOneBit​(initialEntryCnt)+1; ?

Reply:
`2<<Integer.highestOneBit​(initialEntryCnt)+1; ` increases its result too fast, e.g. 
When `initialEntryCnt` is 16, the result has already been 262144 😲

Thanks for your suggestion all the same.

Reply:
ups, I did misread the documentation. Integer.highestOneBit​(initialEntryCnt)*2 then

Reply:
I will tweak the current implementation according to suggestion, which will be more efficient 😄